### PR TITLE
Fix nested for-loop compilation

### LINKED ISSUE #669: SVML compiler miscompiles nested for loops
The following snippet produces the wrong result when run on both the Source §3 Concurrent VM and the Sinter VM:

```js
for (let j = 0; j < 10; j = j + 1) {
    for (let i = 0; i < 1000; i = i + 1) {}
    display(j);
}
```

so I assume it is a miscompilation.

### PR DESCRIPTION
**Pull request by jonchan51**
Fixes #669

### CODE REVIEW

File: src/vm/svml-compiler.ts [hunk 1]
```diff
@@ -1001,20 +1001,16 @@ function transformForLoopsToWhileLoops(program: es.Program) {
       if (init!.type === 'VariableDeclaration') {
         const loopVarName = ((init as es.VariableDeclaration).declarations[0].id as es.Identifier)
           .name
+        const newLoopVarName = loopVarName + '-loop-control-var' // purposely long to reduce unintentional clash
+        const copyOfNewLoopVarName = 'copy-of-' + newLoopVarName
```
Comment (martin-henz) [inline src/vm/svml-compiler.ts:1005]: to be safe you might use a string that is not allowed as a name in JavaScript, maybe

"*" + loopVarName + "*"

Comment (jonchan51) [reply src/vm/svml-compiler.ts:1005]: Forgot to remove the comment, "-" is already an invalid character for identifiers

Comment (martin-henz) [reply src/vm/svml-compiler.ts:1005]: Oh, didn't look close enough. But then why so long?

Comment (martin-henz) [reply src/vm/svml-compiler.ts:1005]: I guess it's nice and clear for debugging. Just remove the comment, then.

Comment (thomastanck) [reply src/vm/svml-compiler.ts:1005]: Probably not a bad thing to have long identifiers. It might also come in handy if manually debugging the VM, maybe.

**Comment by jonchan51** [comment]
Yea, this was a result of mixing renaming and creating new environments for loops. Previously, I did not rename variables within loops because a new environment was being created, which is a decision that didn't make any sense. I changed it such that renaming will happen in loops as well, which should address the issue.

**Pull request merged**